src/modules/items/MegaStoneItem.ts

```diff
@@ -0,0 +1,30 @@
+import { Currency } from '../GameConstants';
+import Item from './Item';
+import { ShopOptions } from './types';
+
+export default class MegaStoneItem extends Item {
+    constructor(private pokemonId: number, megaStoneName: string, basePrice: number, currency: Currency = Currency.questPoint, options?: ShopOptions, displayName?: string) {
```

Comment:
I think we should pass in a PokemonNameType instead of the id, easier to check things are correct.

Reply:
I started with that solution, but finding the image got a bit harder. Then i needed to look up the id somehow...
But if you think it's a better solution, i can try again

Reply:
What do you need the id for? We have `party.getPokemonByName` and `party.alreadyCaughtPokemonByName`

Reply:
oh I see now, the image

Reply:
Do you get errors with `pokemonMap[name].id`?

Reply:
Yeah, that works. Thanks! I have submitted that change now